hs/src/Reach/Connector/ALGO.hs

```diff
@@ -2032,6 +2032,21 @@ ce = \case
     ca oa
     cextract start sz
     cfrombs t
+  DLE_ObjectSet at obj_a fieldName val_a -> do
+    -- This makes the (unsound?) assumption that when adding a new field to an object,
+    -- all copied fields should stay in the order they were already in.
+    let val_t = argTypeOf val_a
+    let oldFieldTypes = argObjstrTypes obj_a
+    let newFieldTypes = M.toAscList $ M.insert fieldName val_t $ M.fromList oldFieldTypes
+    let fieldIdx = fromIntegral $ fromJust $ List.elemIndex fieldName $ map fst newFieldTypes
+    let (_, valStart, oldValLen_) = computeExtract (map snd newFieldTypes) fieldIdx
+    let oldValLen = if length oldFieldTypes == length newFieldTypes then oldValLen_ else 0
```

Comment:
This should just be the size of `val_a`

Reply:
That is only true if we're replacing a field with the exact size as val_a

Reply:
Types all have fixed sizes